src/m_misc.c

```diff
@@ -0,0 +1,20 @@
+/* Copyright (c) 2019 Dan Bornstein.
+* For information on usage and redistribution, and for a DISCLAIMER OF ALL
+* WARRANTIES, see the file, "LICENSE.txt," in this distribution.  */
+
+/* This file has miscellaneous functions that don't have a better home. */
+
+#include "m_pd.h"
+
+/* This compares two floating point values for "strict equality," in the usual
+sense, that is sameness-of-value rather than mathematical-equality. That is,
+unlike `==` which obeys (by and large) IEEE754 rules for equality and hence
+sometimes leads to surprising results (such as the nonreflexivity of NaN, and
+negative and positive zeros being considered equal to each other), this function
+operates on the bitwise representation of its arguments. */
+int float_strict_equal(t_float f1, t_float f2) {
+    /* Unfortunately, the most reliable way to perform the requisite comparison
+    in C is to use `memcmp()`. Fortunately, many compilers understand this
+    particular idiom, so it's typically reasonably performant (just ugly). */
+    return memcmp(&f1, &f2, sizeof (t_float)) == 0;
+}
```

Comment:
This function will yield true if both floats happen to be NaNs with the same bit representation - which is against the IEEE rules. I think you would have to add an explicit NaN check here.

Reply:
Well, that's the thing: In this case I would presume you want two NaNs to be considered the same, so you don't superfluously update the UI.

Reply:
That is, explicitly wanting to _not_ obey IEEE754 math is why (something like) this function is appropriate to use in this context.